### PR TITLE
DOC: initialise random number generator before first use in quickstart

### PR DESCRIPTION
**Pull request by shreepads**
numpy's random number generator is used at several places in the quickstart
but a new user, like me, won't know how to initialise it. Therefore I
recommend adding this at the first place rg is used.

### CODE REVIEW

**Comment by shreepads** [comment]
Hmm the doc tests have failed as the expected random output doesn't match with that produced. I'm not sure how this test could have been working earlier unless rg is being defined somewhere in the test setup phase. If that seed is known I can incorporate it into the rg initialisation

**Comment by mattip** [comment]
The `rng` is initialized in a hidden comment at the top of the file. Perhaps you mean to make that public?

**Comment by shreepads** [comment]
> The `rng` is initialized in a hidden comment at the top of the file. Perhaps you mean to make that public?

Right, I didn't see that! Yes I think it should be made public at the first point of use as it tripped me up while I was going thru the quickstart. Let me try adding that.

Just noticed it is also visibly defined lower down in the 'Histogram' section so should be OK here.

File: doc/source/user/quickstart.rst [hunk 1]
```diff
@@ -348,6 +347,7 @@ existing array rather than create a new one.
 
 ::
 
+    >>> rg = np.random.default_rng(1)
```
Comment (rossbar) [inline doc/source/user/quickstart.rst:350]: Perhaps a comment could be added here as well, to briefly describe what this line does without distracting from the overall theme of the example (illustrating in-place operations)

**Review by rossbar — COMMENTED** [review]
The proposed change is an improvement - having the instantiation of the Generator instance hidden in the `testsetup` directive would be very confusing for users who are not familiar with the new random API.

On a related note, the quickstart tutorial could probably use some new info/text calling attention to the new random API and [the associated docs](https://numpy.org/devdocs/reference/random/index.html). Just a thought, though it should probably be a separate PR since it's a more involved change.

File: doc/source/user/quickstart.rst (hunk 1 above)
Comment (shreepads) [reply doc/source/user/quickstart.rst:350]: Makes sense I'll add a comment (formatted like others on the page) - "create instance of default random number generator"

**Review by WarrenWeckesser — APPROVED** [review]

**Comment by WarrenWeckesser** [comment]
Squashed and merged.  Thanks, @shreepads.

**Comment by shreepads** [comment]
Thanks @WarrenWeckesser ! 
I should celebrate my first teensy contribution to a major open source project but there is a pretty strict lockdown here... Let me go to the... kitchen and get... yeah a cup of coffee. :-)

**Pull request merged**